cmd/list.go

```diff
@@ -44,13 +45,90 @@ const (
 
 	cellSideCount = 2
 
+	headerCount = 4
+
 	paddingHorizontal = 1
 	paddingVertical   = 0
 
 	rowHeaderIdx = 0
 	rowMaxHeight = 1
 )
 
+func getTermInfo() int {
+	width, _, err := term.GetSize(0)
+	if err != nil {
+		return 0
+	}
+
+	return width
+}
+
+func determineColumnWidths(rows ...[]string) (tooWide bool, colWidths []int) {
+	terminalWidth := getTermInfo()
+	idColumnWidth := columnWidthID
+	aliasColumnWidth := columnWidthAlias
+
+	for _, row := range rows {
+		if idWidth := len(row[columnIdxID]); idWidth >= idColumnWidth {
+			idColumnWidth = idWidth + (paddingHorizontal * cellSideCount)
+		}
+
+		if aliasWidth := len(row[columnIdxAlias]); aliasWidth >= aliasColumnWidth {
+			aliasColumnWidth = aliasWidth + (paddingHorizontal * cellSideCount)
+		}
+	}
+
+	totalWidth := (idColumnWidth + aliasColumnWidth + columnWidthVersion + columnWidthNumNodes)
+	totalWidth += headerCount * (paddingHorizontal * cellSideCount)
+
+	return (totalWidth > terminalWidth), []int{idColumnWidth, aliasColumnWidth, columnWidthVersion, columnWidthNumNodes}
+}
+
+func printList(rows ...[]string) {
+	for _, row := range rows {
+		fmt.Fprintf(os.Stdout, "ID: %s\n"+"Alias: %s\n"+"Version: %s\n"+"# Nodes: %s\n\n",
+			row[columnIdxID], row[columnIdxAlias], row[columnIdxVersion], row[columnIdxNumNodes])
+	}
+}
+
+func printTable(colWidths []int, rows ...[]string) {
```

Comment:
Same comment as above about printing directly in the called function, rename to `formatTable` and return a string to print from the calling context